### ISSUE TITLE
Uncaught TypeError: history.getChangesSinceCheckpoint is not a function

### ISSUE THREAD

**Issue by andypayne** [t9md/atom-vim-mode-plus#288]
1. Create a new file in Atom (Cmd-n).
2. Switch to another application and copy some text. For an example that reproduced for me, I copied: "To edit keymap edit your keymap.cson" (without quotes) from a web page in Chrome. However it seems not to be particular about the copied text.
3. Switch to Atom.
4. Switch to insert mode (i).
5. Paste the text with Cmd-v.
6. Leave insert mode (Esc). This triggers the error.

**Atom Version**: 1.8.0-beta2
**System**: Mac OS X 10.11
**Thrown From**: [vim-mode-plus](https://github.com/t9md/atom-vim-mode-plus) package, v0.43.0

### Stack Trace

Uncaught TypeError: history.getChangesSinceCheckpoint is not a function

```
At /Users/andpayne/.atom/packages/vim-mode-plus/lib/utils.coffee:155

TypeError: history.getChangesSinceCheckpoint is not a function
    at getNewTextRangeFromCheckpoint (/Users/andpayne/.atom/packages/vim-mode-plus/lib/utils.coffee:155:22)
    at /Users/andpayne/.atom/packages/vim-mode-plus/lib/operator.coffee:1076:10
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at ModeManager.deactivate (/Users/andpayne/.atom/packages/vim-mode-plus/lib/mode-manager.coffee:68:14)
    at ModeManager.activate (/Users/andpayne/.atom/packages/vim-mode-plus/lib/mode-manager.coffee:48:6)
    at VimState._results.push._this.(anonymous function) [as activate] (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/text-buffer/node_modules/delegato/lib/delegator.js:67:61)
    at VimState.commands.activate-normal-mode (/Users/andpayne/.atom/packages/vim-mode-plus/lib/main.coffee:133:35)
    at atom-text-editor.<anonymous> (/Users/andpayne/.atom/packages/vim-mode-plus/lib/main.coffee:164:16)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:260:29)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:580:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:388:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:98:36)
    at HTMLDocument.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:3:61)
```

### Commands

```
     -0:21.8.0 vim-mode-plus:reset-normal-mode (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
  2x -0:21.6.0 vim-mode-plus:move-up (atom-text-editor.editor.vim-mode-plus.is-focused.normal-mode)
     -0:20.5.0 vim-mode-plus:move-to-beginning-of-line (atom-text-editor.editor.vim-mode-plus.is-focused.normal-mode)
     -0:20.4.0 vim-mode-plus:move-down (atom-text-editor.editor.vim-mode-plus.is-focused.normal-mode)
     -0:20.1.0 vim-mode-plus:move-up (atom-text-editor.editor.vim-mode-plus.is-focused.normal-mode)
  2x -0:19.8.0 vim-mode-plus:delete (atom-text-editor.editor.vim-mode-plus.is-focused.normal-mode)
  2x -0:19.4.0 vim-mode-plus:move-up (atom-text-editor.editor.vim-mode-plus.is-focused.normal-mode)
     -0:19 vim-mode-plus:move-down (atom-text-editor.editor.vim-mode-plus.is-focused.normal-mode)
  2x -0:18.6.0 vim-mode-plus:delete (atom-text-editor.editor.vim-mode-plus.is-focused.normal-mode)
 11x -0:18.4.0 vim-mode-plus:move-down (atom-text-editor.editor.vim-mode-plus.is-focused.normal-mode)
 12x -0:17.6.0 vim-mode-plus:move-up (atom-text-editor.editor.vim-mode-plus.is-focused.normal-mode)
 13x -0:16.9.0 vim-mode-plus:move-down (atom-text-editor.editor.vim-mode-plus.is-focused.normal-mode)
     -0:15.1.0 application:new-file (atom-text-editor.editor.vim-mode-plus.is-focused.normal-mode)
     -0:01.4.0 vim-mode-plus:activate-insert-mode (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
     -0:01.0 core:paste (atom-text-editor.editor.vim-mode-plus.is-focused.insert-mode)
     -0:00.5.0 vim-mode-plus:activate-normal-mode (atom-text-editor.editor.vim-mode-plus.is-focused.insert-mode)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "ex-mode",
      "vim-surround"
    ],
    "themes": [
      "one-dark-ui",
      "atom-dark-syntax"
    ]
  },
  "vim-mode-plus": {
    "useSmartcaseForSearch": true
  }
}
```

### Installed Packages

```coffee
# User
clipboard-plus, v0.5.1
react, v0.15.0
vim-mode-clipboard-plus, v0.2.0
vim-mode-plus, v0.43.0

# Dev
No dev packages
```

**Comment by t9md** [comment]
It seems its not likely to bug of vim-mode-plus.
history.getChangesSinceCheckpoint is function of text-buffer and history is preserved by
`{history} = editor.getBuffer()`.

This function should be always available.
So not sure direct cause.

Do you reproduce it 100% in your environment(I tried but I couldn't in my 1.8.0-beta4)?.
If you can reproduce it 100%, always, updating latest beta(=1.8.0-beta4) solve this problem?

**Comment by notauni** [comment]
I'm getting the same error without needing to copy files. 

1. Open a new file in atom (ctrl + n)
2. Switch to insert mode (i)
3. Exit insert mode (esc)
4. Error popup

**Atom Version**: 1.7.2
**System**: Ubuntu 14.04.4
**Thrown From**: [vim-mode-plus](https://github.com/t9md/atom-vim-mode-plus) package, v0.43.0


### Stack Trace

Uncaught TypeError: history.getChangesSinceCheckpoint is not a function

```
At /home/notauni/.atom/packages/vim-mode-plus/lib/utils.coffee:145

TypeError: history.getChangesSinceCheckpoint is not a function
    at getNewTextRangeFromCheckpoint (/home/notauni/.atom/packages/vim-mode-plus/lib/utils.coffee:145:22)
    at /home/notauni/.atom/packages/vim-mode-plus/lib/operator.coffee:1085:10
    at Function.module.exports.Emitter.simpleDispatch (/usr/share/atom/resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/usr/share/atom/resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at ModeManager.deactivate (/home/notauni/.atom/packages/vim-mode-plus/lib/mode-manager.coffee:71:14)
    at ModeManager.activate (/home/notauni/.atom/packages/vim-mode-plus/lib/mode-manager.coffee:51:6)
    at VimState._results.push._this.(anonymous function) [as activate] (/usr/share/atom/resources/app.asar/node_modules/text-buffer/node_modules/delegato/lib/delegator.js:67:61)
    at VimState.commands.activate-normal-mode (/home/notauni/.atom/packages/vim-mode-plus/lib/main.coffee:131:35)
    at atom-text-editor.<anonymous> (/home/notauni/.atom/packages/vim-mode-plus/lib/main.coffee:162:16)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/usr/share/atom/resources/app.asar/src/command-registry.js:260:29)
    at /usr/share/atom/resources/app.asar/src/command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/usr/share/atom/resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:580:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/usr/share/atom/resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:388:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (/usr/share/atom/resources/app.asar/src/window-event-handler.js:98:36)
    at HTMLDocument.<anonymous> (/usr/share/atom/resources/app.asar/src/window-event-handler.js:3:61)
```

### Commands

```
     -0:02.7.0 application:new-file (div.pane-item.native-key-bindings.about)
     -0:00.9.0 vim-mode-plus:activate-insert-mode (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
     -0:00.5.0 vim-mode-plus:activate-normal-mode (atom-text-editor.editor.vim-mode-plus.is-focused.insert-mode)
```

### Config

```json
{
  "core": {}
}
```

### Installed Packages

```coffee
# User
file-icons, v1.7.13
minimap, v4.24.3
vim-mode-plus, v0.43.0

# Dev
No dev packages
```

**Comment by t9md** [comment]
Thanks for reporting @notauni.
I checked text-buffer code quickly, as [this line](https://github.com/atom/text-buffer/blob/master/src/text-buffer.coffee#L661) indicates, it seems there is some situation where buffer.history is `null`.

So I will simply add null guard to fix this issue.

**Comment by t9md** [comment]
Released v0.44.0, reopen if it still happen.
